### PR TITLE
Fix KVM cloudstack-agent start if there is vm not managed by cloudstack on the host

### PR DESCRIPTION
**Pull request by harikrishna-patnala**
### Description

This PR fixes the issue #8040 

we are now handling any exception while setting up the memory balloon stats period

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by harikrishna-patnala** [comment]
@blueorangutan package

**Comment by harikrishna-patnala** [comment]
@blueorangutan package

**Review by weizhouapache — APPROVED** [review]
code lgtm

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java [hunk 1]
```diff
@@ -1313,8 +1313,8 @@ protected void setupMemoryBalloonStatsPeriod(Connect conn) {
                 }
                 s_logger.debug(String.format("The memory balloon stats period [%s] has been set successfully for the VM (Libvirt Domain) with ID [%s] and name [%s].",
                         currentVmBalloonStatsPeriod, vmId, dm.getName()));
-            } catch (final LibvirtException e) {
-                s_logger.warn("Failed to set up memory balloon stats period." + e.getMessage());
+            } catch (final Exception e) {
```
Comment (DaanHoogland) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:1316]: Is this needed or can we add a specific Exception to the catch clause?

Comment (harikrishna-patnala) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:1316]: @DaanHoogland we are trying to read many other parameters in that parser. so there are chances of other errors such as NPEs, cast errors. While discussing with Wei several other found in IDE, so thought a common exception could catch it and warn the message. 
image

Comment (DaanHoogland) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:1316]: ok, I hate it and I would like to see a message per exception, but no -1 for that

**Comment by GutoVeronezi** [comment]
Should we consider VMs not managed by ACS in Agent start up? Would not it be easier to filter out those VMs?

**Comment by weizhouapache** [comment]
> Should we consider VMs not managed by ACS in Agent start up? Would not it be easier to filter out those VMs?

@GutoVeronezi 
we had some discussion on the issue https://github.com/apache/cloudstack/issues/8040
the problem is how to accurately determine if a vm is managed on ACS or not in kvm agent (which cannot  read cloudstack database), especially considering that the kvm import feature will be introduced in 4.19

**Comment by GutoVeronezi** [comment]
> > Should we consider VMs not managed by ACS in Agent start up? Would not it be easier to filter out those VMs?
> 
> @GutoVeronezi we had some discussion on the issue #8040 the problem is how to accurately determine if a vm is managed on ACS or not in kvm agent (which cannot read cloudstack database), especially considering that the kvm import feature will be introduced in 4.19

I see, thanks @weizhouapache.

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java (hunk 1 above)
Comment (harikrishna-patnala) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:1316]: :) So you are giving +1 on this right ;)

**Comment by weizhouapache** [comment]
@blueorangutan package

**Review by DaanHoogland — COMMENTED** [review]
+0

**Comment by harikrishna-patnala** [comment]
@DaanHoogland do you want me to improve this code to handle specific exceptions ?

**Comment by DaanHoogland** [comment]
> @DaanHoogland do you want me to improve this code to handle specific exceptions ?

I would prefer that but am not going to -1 this PR because of it.

**Review by rohityadavcloud — APPROVED** [review]
LGTM - didn't test it though

**Comment by weizhouapache** [comment]
@DaanHoogland @harikrishna-patnala 
I think it is not a problem if ACS fails to set the memory ballooning of a specific VM, we could silently ignore the error, no matter what exceptions (LibvirtException, NPE, or others) are thrown.

**Comment by DaanHoogland** [comment]
> @DaanHoogland @harikrishna-patnala I think it is not a problem if ACS fails to set the memory ballooning of a specific VM, we could silently ignore the error, no matter what exceptions (LibvirtException, NPE, or others) are thrown.

ok

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan LLtest matrix basicZone

**Comment by DaanHoogland** [comment]
@blueorangutan LLtest basicZone

**Comment by DaanHoogland** [comment]
@blueorangutan LLtest securityGroups

**Comment by DaanHoogland** [comment]
@blueorangutan test rocky8 kvm-ol9

**Comment by DaanHoogland** [comment]
@blueorangutan test rocky8 kvm-alma9

**Comment by DaanHoogland** [comment]
@blueorangutan test rocky8 kvm-rocky8

**Comment by DaanHoogland** [comment]
smoke tests for security groups and basic zone are expected to have such a low pass rate.
regular smoke tests look fine

**Review by DaanHoogland — APPROVED** [review]

**Review by rajujith — APPROVED** [review]
LGTM. Observed a similar issue parsing domain XML for a VM created outside Cloudstack with 4.18.1, and confirmed this PR fixes the issue.

```
2023-11-03 06:43:58,568 ERROR [cloud.agent.AgentShell] (main:null) (logid:) Unable to start agent:
javax.naming.ConfigurationException: Error while creating Agent with class [com.cloud.hypervisor.kvm.resource.LibvirtComputingResource]. [Root exception is java.lang.IllegalArgumentException: No enum constant com.cloud.hypervisor.kvm.resource.LibvirtVMDef.ChannelDef.ChannelType.SPICEVMC]
	at com.cloud.agent.AgentShell.launchAgentFromClassInfo(AgentShell.java:434)
	at com.cloud.agent.AgentShell.launchAgent(AgentShell.java:415)
	at com.cloud.agent.AgentShell.start(AgentShell.java:511)
	at com.cloud.agent.AgentShell.main(AgentShell.java:541)
Caused by: java.lang.IllegalArgumentException: No enum constant com.cloud.hypervisor.kvm.resource.LibvirtVMDef.ChannelDef.ChannelType.SPICEVMC
	at java.base/java.lang.Enum.valueOf(Enum.java:240)
	at com.cloud.hypervisor.kvm.resource.LibvirtVMDef$ChannelDef$ChannelType.valueOf(LibvirtVMDef.java:1768)
	at com.cloud.hypervisor.kvm.resource.LibvirtDomainXMLParser.parseDomainXML(LibvirtDomainXMLParser.java:273)
	at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.setupMemoryBalloonStatsPeriod(LibvirtComputingResource.java:1301)
	at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.configure(LibvirtComputingResource.java:1244)
	at com.cloud.agent.Agent.<init>(Agent.java:190)
	at com.cloud.agent.AgentShell.launchNewAgent(AgentShell.java:452)
	at com.cloud.agent.AgentShell.launchAgentFromClassInfo(AgentShell.java:431)
	... 3 more


```

**Pull request merged**